base/strings/basic.jl

```diff
@@ -269,11 +285,32 @@ function prevind(s::AbstractString, i::Integer)
     return 0 # out of range
 end
 
+function prevind(s::AbstractString, i::Integer, nchar::Integer)
+    nchar > 0 || throw(ArgumentError("nchar must be greater than 0"))
```

Comment:
Why not accept `0`?

Reply:
Handling of `0`:
* requires separate logic (additional `if`)
* is unclear what should be the result if `i` is not a proper byte index (if one calls `prevind` one expects to get a proper byte index in return - this is an important invariant of those functions in my opinion)